cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1168,7 +1168,7 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                                         + "' does not exist in namespace " + clusterNamespace + ".")).mapEmpty();
                     } else if (kafka.getStatus() == null
                             || kafka.getStatus().getConditions() == null
-                            || kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) {
+                            || kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True")) && state(kafkaRebalance) != KafkaRebalanceState.Ready) {
```

Comment:
I think we are missing opening and closing parenthesis to group the "big" OR across the Kafka related status.
The && is just added at the end but because of the boolean operators precedence it's evaluated first so:

`true || false && false ===> true`
`(true || false) && false ===> false`

So I think we should factoring out a method like `isKafkaClusterReady` which evaluate the big chain of OR we have, and then having something like:

```java
if (!isKafkaClusterReady() && state(kafkaRebalance) != KafkaRebalanceState.Ready) {
...
}
```

@fvaleri @ShubhamRwt wdyt?

Reply:
@ppatierno I agree with you. The changes are missing the opening and closing parenthesis to group the `OR` which is not correct. Yeah it would be great to have a method to avoid a big `OR` but if its not required then we should at least fix the parenthesis issue

Reply:
@ppatierno you are absolutely right. 

I agree with the refactoring you propose, in fact I missed the grouping because it's hard to read. Let me add that the test should cover all permutations.